internal/authn/oidc/oidc.go

```diff
@@ -97,23 +96,27 @@ func issuerCheck(issuer string, domain string) bool {
 	domain = strings.ToLower(domain)
 	issuer = strings.ToLower(issuer)
 
-	// Define a regex pattern to strip "http://", "https://", and any "/path-uri" part from the Issuer
-	re := regexp.MustCompile(`^(https?://)?([^/]+)`)
+	if strings.HasPrefix(issuer, "http://") {
+		issuer = issuer[7:]
+	} else if strings.HasPrefix(issuer, "https://") {
+		issuer = issuer[8:]
+	}
 
-	// Extract the host part from the input
-	match := re.FindStringSubmatch(issuer)
-	if len(match) < 3 {
-		return false
+	if colonIndex := strings.Index(issuer, ":"); colonIndex != -1 {
+		issuer = issuer[:colonIndex]
 	}
-	// The actual host without the scheme and path
-	actualIssuer := strings.ToLower(match[2])
 
-	if domain == actualIssuer {
-		return true
+	if slashIndex := strings.Index(issuer, "/"); slashIndex != -1 {
+		issuer = issuer[:slashIndex]
 	}
 
-	if strings.HasSuffix(actualIssuer, "."+domain) {
+	if dotIndex := strings.LastIndex(issuer, "."); dotIndex != -1 {
+		issuer = issuer[:dotIndex]
+	}
+
+	if strings.HasSuffix(issuer, "."+domain) {
 		return true
 	}
-	return false
+
+	return issuer == domain
 }
```

Comment:
Could we add some tests against this issuerCheck?

Reply:
remove the issuer check